src/commands.ts

```diff
@@ -54,15 +54,19 @@ const digits = keyValueBy(Array(58 - 48).fill(0), (n, i) => ({
   [48 + i]: i.toString(),
 }))
 
-/** Hash all the properties of a command into a string that can be compared with the result of hashKeyDown. */
-export const hashCommand = (command: Command): string => {
-  const keyboard = typeof command.keyboard === 'string' ? { key: command.keyboard } : command.keyboard || ({} as Key)
-  return (
-    (keyboard.meta ? 'META_' : '') +
-    (keyboard.alt ? 'ALT_' : '') +
-    (keyboard.shift ? 'SHIFT_' : '') +
-    keyboard.key?.toUpperCase()
-  )
+/** Hash all the properties of a command into an array of strings that can be compared with the result of hashKeyDown. */
+export const hashCommand = (command: Command): string[] => {
```

Comment:
One thing to keep in mind is that when you hash something you turn it into a string. So the names are a bit muddled now with `hashCommand` incongruously returning an array of strings. Furthermore, the comments are now confusing, as they say the result of `hashCommand` and `hashKeyDown` can be directly compared, but that is no longer the case. 

Instead, I would suggest preserving `hashCommand` as it is, and adding the array-handling logic to `index()`. `hashCommand` and `hashKeyDown` are a logical pair and we want to preserve that symmetry. Unfortunately Typescript cannot capture the fact that they both return the same subset of `string`.